tests/components/plex/test_init.py

```diff
@@ -131,45 +148,40 @@ def __init__(self):
     )
 
     # Test with account failure
-    with patch(
-        "plexapi.server.PlexServer", side_effect=WrongCertHostnameException
-    ), patch(
-        "plexapi.myplex.MyPlexAccount", side_effect=plexapi.exceptions.Unauthorized
-    ):
-        old_entry.add_to_hass(hass)
-        assert await hass.config_entries.async_setup(old_entry.entry_id) is False
-        await hass.async_block_till_done()
+    requests_mock.get(f"{old_url}/accounts", status_code=401)
+    old_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(old_entry.entry_id) is False
+    await hass.async_block_till_done()
 
     assert old_entry.state == ENTRY_STATE_SETUP_ERROR
     await hass.config_entries.async_unload(old_entry.entry_id)
 
     # Test with no servers found
-    with patch(
-        "plexapi.server.PlexServer", side_effect=WrongCertHostnameException
-    ), patch("plexapi.myplex.MyPlexAccount", return_value=MockPlexAccount(servers=0)):
-        assert await hass.config_entries.async_setup(old_entry.entry_id) is False
-        await hass.async_block_till_done()
+    requests_mock.get(f"{old_url}/accounts", text=PMS_ACCOUNT_PAYLOAD)
+    requests_mock.get("https://plex.tv/api/resources", text=EMPTY_PAYLOAD)
+
+    assert await hass.config_entries.async_setup(old_entry.entry_id) is False
+    await hass.async_block_till_done()
 
     assert old_entry.state == ENTRY_STATE_SETUP_ERROR
     await hass.config_entries.async_unload(old_entry.entry_id)
 
     # Test with success
-    with patch(
-        "plexapi.server.PlexServer", side_effect=WrongCertHostnameException
-    ), patch("plexapi.myplex.MyPlexAccount", return_value=MockPlexAccount()):
-        assert await hass.config_entries.async_setup(old_entry.entry_id)
-        await hass.async_block_till_done()
+    new_url = PLEX_DIRECT_URL
```

Comment:
What causes this block to start succeeding?    That is, how does the plex client know to go from requesting the old url to request the new url?

Reply:
It finds the proper Plex server from the `/resources` endpoint in the next line. The previous step in the test doesn't return any valid Plex servers in the mocked payload so we unload and try again.

Reply:
OK, i think i was confused because the start of the test also has this line too:
```
requests_mock.get("https://plex.tv/api/resources", text=PLEXTV_RESOURCES)
```

But then i guess it changes back and forth a few times between different types of failure modes.

Is there a benefit of testing these in a single long test case to test recovery between these different failure modes -- or could each of these failure modes be split into their own focused test? That might make it easier to compare and contrast between the test setup methods since there won't be any inherited state.